components/script/dom/node.rs

```diff
@@ -1230,66 +1271,76 @@ impl Iterator for PrecedingNodeIterator {
         } else if let Some(previous_sibling) = current.GetPreviousSibling() {
             if self.root == previous_sibling {
                 None
-            } else if let Some(last_child) = previous_sibling.descending_last_children().last() {
+            } else if let Some(last_child) = previous_sibling.descending_last_children::<Node>().last() {
                 Some(last_child)
             } else {
                 Some(previous_sibling)
             }
         } else {
             current.GetParentNode()
         };
-        self.current.clone()
+
+        //TODO: dominotree - not sure how to handle failures here - is returning None okay if it can't cast to T?
```

Comment:
If it is ok, you should call `unwrap` to assert that it is indeed ok, or refactor the whole method to avoid that downcast here.